### PR TITLE
ci: Update CI workflows

### PR DESCRIPTION
**Pull request by pgrzesik**
Changes:
- Introduce Integrate CI Workflow
- Introduce new publish Workflow
- Introduce commitlint
- Introduce changelog generation
- Update validate workflow

### CODE REVIEW

**Comment by pgrzesik** [comment]
After merging, I will change the required CI checks in project settings

File: .github/dependabot.yml [hunk 1]
```diff
@@ -1,12 +1,12 @@
 version: 2
```
Comment (medikoo) [inline .github/dependabot.yml:1]: I will remove this file. I think it's better to just bump deps before release, then accepting automatically created PR for each. WDYT?

File: .github/workflows/integrate.yml [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+        run: |
+          npm update --no-save
+          npm update --save-dev --no-save
+      - name: Unit tests
+        run: script -e -c "npm test"
+
+  tagIfNewVersion:
+    name: Tag if new version
+    runs-on: ubuntu-latest
+    needs: [linuxNode14, windowsNode14, linuxNode16, linuxNode12]
```
Comment (medikoo) [inline .github/workflows/integrate.yml:220]: Job names need to be updated here

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -1,18 +1,52 @@
...
+        uses: actions/checkout@v2
+
+      - name: Retrieve node_modules from cache
+        id: cacheNodeModules
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.npm
+            node_modules
+          key: npm-v14-${{ runner.os }}-refs/heads/master-${{ hashFiles('package.json') }}
```
Comment (medikoo) [inline .github/workflows/publish.yml:29]: If in others we treat v16 as prime version, I would stick to v16 here as well

File: .github/workflows/validate.yml [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+
+      - name: Install dependencies
+        if: steps.cacheNpm.outputs.cache-hit != 'true'
+        run: |
+          npm update --no-save
+          npm update --save-dev --no-save
+      - name: Unit tests
+        # Some tests depend on TTY support, which is missing in GA runner
+        # Workaround taken from https://github.com/actions/runner/issues/241#issuecomment-577360161
+        run: script -e -c "npm test"
```
Comment (medikoo) [inline .github/workflows/validate.yml:197]: This form is needed in `serverless` repository, but is it needed here? Or maybe we can just do `npm test`

File: .github/dependabot.yml (hunk 1 above)
Comment (pgrzesik) [reply .github/dependabot.yml]: I think we can remove it :+1: I've added commit that removes it

File: .github/workflows/integrate.yml (hunk 2 above)
Comment (pgrzesik) [reply .github/workflows/integrate.yml]: good catch :+1:

**Pull request merged**